stdpopsim/genomes.py

```diff
@@ -386,6 +386,70 @@ def add_dfe(self, intervals, DFE):
         self.dfe_list.append(DFE)
         self.interval_list.append(intervals)
 
+    def add_single_site_mutation_type(
+        self, id, selection_coeff=0.0, dominance_coeff=1.0
+    ):
+        """
+        Adds a DFE representing a mutation at a single site, contain a single
+        :class:`MutationType` with the provided parameters. The string ID of
+        the mutation may be referenced by extended events passed to the
+        simulation engine.
+
+        For instance, to simulate a selective sweep,
+
+        .. code-block:: python
+
+            contig.add_single_site_mutation_type(
+                id="sweep_variant",
+                selection_coef=0.1,
+            )
+            extended_events = [
+                stdpopsim.ext.DrawMutation(
+                    time=mutation_time,
+                    mutation_id="sweep_variant",
+                    population_id=0,
+                    coordinate=mutation_coordinate,
+                    save=True,
+                ),
+                stdpopsim.ext.ConditionOnAlleleFrequency(
+                    start_time=stdpopsim.ext.GenerationAfter(mutation_time),
+                    end_time=0,
+                    mutation_id="sweep_variant",
+                    population_id=0,
+                    op=">",
+                    allele_frequency=0.0,
+                ),
+            ]
+            engine = stdpopsim.get_engine("slim")
+            ts_sweep = engine.simulate(
+                ...,
+                extended_events=extended_events,
+            )
```

Comment:
so I guess next step is to write a helper function for this code block to ease the sweep simulation interface?